addons/mail/models/mail_channel.py

```diff
@@ -294,7 +294,7 @@ def _subscribe_users_automatically_get_members(self):
         """ Return new members per channel ID """
         return dict(
             (channel.id, (channel.group_ids.users.partner_id - channel.channel_partner_ids).ids)
-            for channel in self
+            for channel in self.with_context(active_test=False)
```

Comment:
Hello,

I believe this will have the side effect to automatically add archived users to the channel, which was not the case before.

We might want to limit the context to the `channel_partner_ids` part?

Also please add a test to cover the fix.

Thanks!